rust/decentraland-godot-lib/src/godot_classes/dcl_ui_control.rs

```diff
@@ -0,0 +1,173 @@
+};
+
+#[derive(GodotClass)]
+#[class(base=Control)]
+pub struct DclUiControl {
+    #[base]
+    base: Base<Control>,
+
+    #[export]
+    dcl_entity_id: SceneEntityId,
+
+    ui_result: Option<Rc<RefCell<UiResults>>>,
+
+    is_gui_input_signal_connected: bool,
+    force_pointer_filter_mode: PointerFilterMode,
+
+    listening_mouse_down: bool,
+    listening_mouse_up: bool,
+}
+
+#[godot_api]
+impl NodeVirtual for DclUiControl {
+    fn init(base: Base<Control>) -> Self {
+        Self {
+            base,
+            is_gui_input_signal_connected: false,
+            force_pointer_filter_mode: PointerFilterMode::PfmNone,
+            listening_mouse_down: false,
+            listening_mouse_up: false,
+            ui_result: None,
+            dcl_entity_id: SceneEntityId::ROOT,
+        }
+    }
+
+    fn ready(&mut self) {
+        self.base.set_focus_mode(FocusMode::FOCUS_NONE);
+    }
+}
+
+#[godot_api]
+impl DclUiControl {
+    pub fn new_alloc() -> Gd<Self> {
+        Gd::new_default()
+    }
+
+    #[func]
+    pub fn _on_gui_input(&mut self, input: Gd<InputEvent>) {
+        let global_tick_number = unsafe { *GLOBAL_TICK_NUMBER.as_ptr() as u32 };
+        if let Some(event) = input.try_cast::<InputEventMouseButton>() {
+            let is_left_button = event.get_button_index() == MouseButton::MOUSE_BUTTON_LEFT;
+            let down_event = event.is_pressed();
+
+            if self.listening_mouse_down && is_left_button && down_event {
+                if let Some(ui_result) = self.ui_result.as_ref() {
+                    ui_result.borrow_mut().pointer_event_results.push((
+                        self.dcl_entity_id,
+                        PbPointerEventsResult {
+                            button: InputAction::IaPointer as i32,
+                            hit: None,
+                            state: PointerEventType::PetDown as i32,
+                            timestamp: global_tick_number,
+                            analog: None,
+                            tick_number: global_tick_number,
+                        },
+                    ));
+                }
+            } else if self.listening_mouse_up && is_left_button && !down_event {
+                if let Some(ui_result) = self.ui_result.as_ref() {
+                    ui_result.borrow_mut().pointer_event_results.push((
+                        self.dcl_entity_id,
+                        PbPointerEventsResult {
+                            button: InputAction::IaPointer as i32,
+                            hit: None,
+                            state: PointerEventType::PetUp as i32,
+                            timestamp: global_tick_number,
+                            analog: None,
+                            tick_number: global_tick_number,
+                        },
+                    ));
+                }
+            }
+        }
+
+        // TODO: it enables HOVER and LEAVE events
```

Comment:
What is the status of this comment code?

Reply:
It isn't specified in the ADR and not implemented in the SDK, if we add it, no one could use it